packages/sitecore-jss-react/src/components/FEaaSComponent.tsx

```diff
@@ -1,94 +1,119 @@
-/* eslint-disable react/prop-types */
-import React, { useEffect } from 'react';
-import { ComponentParams, ComponentFields, getFieldValue } from '@sitecore-jss/sitecore-jss/layout';
-// eslint-disable-next-line
-import type * as _FEAAS from '@sitecore-feaas/clientside';
+import React from 'react';
+import * as FEAAS from '@sitecore-feaas/clientside/react';
+import { ComponentFields, getFieldValue } from '@sitecore-jss/sitecore-jss/layout';
 
 export const FEAAS_COMPONENT_RENDERING_NAME = 'FEaaSComponent';
-export const FEAAS_MODULE_SRC = 'https://feaasstatic.blob.core.windows.net/packages/clientside/latest/browser/index.esm.js';
 
-export type FEaaSComponentParams = ComponentParams & {
+export type FEaaSComponentParams = {
   LibraryId?: string;
   ComponentId?: string;
   ComponentVersion?: string;
-  ComponentRevision?: string;
+  ComponentRevision?: number | 'staged' | 'published' | 'saved';
   ComponentHostName?: string;
   ComponentInstanceId?: string;
   ComponentDataOverride?: string;
   ComponentHTMLOverride?: string;
+  styles?: string;
+  RenderingIdentifier?: string;
 };
 
+/**
+ * Props for FEaaS Component
+ * @param {string} template HTML template for presentation rendered inside the component
+ * @param {string} lastModified the date component data was last modified
+ * @param {string} src component endpoint URL
+ * @param {FEaaSComponentParams} params parameters from Sitecore's FEAAS component
+ * @param {ComponentFields} fields field data from component's datasource 
+ */
 export type FEaaSComponentProps = {
+  template: string;
+  lastModified: string;
+  src?: string;
```

Comment:
Should `src` really be a prop? Or could it be generated wherever we need it?

Reply:
I think it should stay. We would always pass params through FEaaSWrapper which can be used to compose the src, but when Wrapper is not used, FEaaSComponent can also be rendered with src only on its own.